src/Features/CSharp/Portable/SplitOrMergeIfStatements/CSharpIfLikeStatementGenerator.cs

```diff
@@ -0,0 +1,151 @@
+            return false;
+        }
+
+        public SyntaxNode GetCondition(SyntaxNode ifLikeStatement)
+        {
+            var ifStatement = (IfStatementSyntax)ifLikeStatement;
+            return ifStatement.Condition;
+        }
+
+        public SyntaxNode GetRootIfStatement(SyntaxNode ifLikeStatement)
+        {
+            var ifStatement = (IfStatementSyntax)ifLikeStatement;
+
+            while (ifStatement.Parent is ElseClauseSyntax elseClause)
+            {
+                ifStatement = (IfStatementSyntax)elseClause.Parent;
+            }
+
+            return ifStatement;
+        }
+
+        public ImmutableArray<SyntaxNode> GetElseLikeClauses(SyntaxNode ifLikeStatement)
+        {
+            var ifStatement = (IfStatementSyntax)ifLikeStatement;
+
+            var builder = ImmutableArray.CreateBuilder<SyntaxNode>();
+
+            while (ifStatement.Else?.Statement is IfStatementSyntax elseIfStatement)
+            {
+                builder.Add(elseIfStatement);
+                ifStatement = elseIfStatement;
+            }
+
+            if (ifStatement.Else != null)
+            {
+                builder.Add(ifStatement.Else);
+            }
+
+            return builder.ToImmutable();
+        }
+
+        public SyntaxNode WithCondition(SyntaxNode ifLikeStatement, SyntaxNode condition)
+        {
+            var ifStatement = (IfStatementSyntax)ifLikeStatement;
+            return ifStatement.WithCondition((ExpressionSyntax)condition);
+        }
+
+        public SyntaxNode WithStatement(SyntaxNode ifLikeStatement, SyntaxNode statement)
+        {
+            var ifStatement = (IfStatementSyntax)ifLikeStatement;
+            return ifStatement.WithStatement(SyntaxFactory.Block((StatementSyntax)statement));
+        }
+
+        public SyntaxNode WithStatementsOf(SyntaxNode ifLikeStatement, SyntaxNode otherIfLikeStatement)
+        {
+            var ifStatement = (IfStatementSyntax)ifLikeStatement;
+            var otherIfStatement = (IfStatementSyntax)otherIfLikeStatement;
+            return ifStatement.WithStatement(otherIfStatement.Statement);
+        }
+
+        public SyntaxNode WithElseLikeClausesOf(SyntaxNode ifStatement, SyntaxNode otherIfStatement)
+        {
+            return ((IfStatementSyntax)ifStatement).WithElse(((IfStatementSyntax)otherIfStatement).Else);
+        }
+
+        public SyntaxNode ToIfStatement(SyntaxNode ifLikeStatement)
+            => ifLikeStatement;
+
+        public SyntaxNode ToElseIfClause(SyntaxNode ifLikeStatement)
+            => ((IfStatementSyntax)ifLikeStatement).WithElse(null);
+
+        public void InsertElseIfClause(SyntaxEditor editor, SyntaxNode afterIfLikeStatement, SyntaxNode elseIfClause)
+        {
+            var ifStatement = (IfStatementSyntax)afterIfLikeStatement;
+            var elseIfStatement = (IfStatementSyntax)elseIfClause;
+
+            var newElseIfStatement = elseIfStatement.WithElse(ifStatement.Else);
```

Comment:
perhaps assert that the elseIf doesn't already have an 'else'?  or is that a supported cases?

Reply:
temporary not really needed.  consider just inlining below.

Reply:
It can have an else, but the idea/"abstraction" of these helpers is that an else-if clause is independent of any other else-if or else clauses, like the way it's represented in VB. So if somebody takes an else-if and inserts is somewhere else, it's fine that its else is dropped, that's the expected behavior.